### ISSUE TITLE
CB unrestricted region/gender

### ISSUE THREAD

**Issue by z3rofeels** [0xboobface/ctbrec#60]
Is it possible to record models that restrict region/gender while also being logged into main account? I can add them manually and they show online but won't record.

**Comment by 0xboobface** [comment]
I have no idea what you mean. Can be explain a bit more what the problem is? I have never seen region or gender restrictions.

**Comment by Asynchr0nous** [comment]
some models block people from accessing their room if they're from certain country/region e.g. a model might block everyone living say just new york yet allow everyone else access her room. So what he's asking if for support for proxy to bypass this

**Comment by z3rofeels** [comment]
Chaturbate allows models to restrict access based on gender/region. If you try to access cams that don't allow your gender/region you're greeted with "Access Denied. This room is not available to your region or gender." There are scripts to bypass the restriction and let you watch/record anonymously (not logged in). If I log in to my main account on CTB, manually adding a model that has restrictions will show them online and added but won't record unless I go to the site and change my gender or location. Doing that then conflicts with the people who have opposite restrictions.

**Comment by z3rofeels** [comment]
> some models block people from accessing their room if they're from certain country/region e.g. a model might block everyone living say just new york yet allow everyone else access her room. So what he's asking if for support for proxy to bypass this

Thanks for explaining it in simpler terms than I could.

**Comment by 0xboobface** [comment]
Yeah, that's what I figured. They most certainly use geoblocking. Otherwise everyone could just change their settings and watch the stream anyway.
So, the best solution still is a VPN or proxy server in a country, which isn't restricted.

**Comment by z3rofeels** [comment]
> Yeah, that's what I figured. They most certainly use geoblocking. Otherwise everyone could just change their settings and watch the stream anyway.
> So, the best solution still is a VPN or proxy server in a country, which isn't restricted.

Actually, you CAN just change your settings and it works.  More performers block gender than region. It usually applies to straight performers who don't want gay watchers and vice versa. If we could log into two accounts (one account as male, one account as female), then there would be no issues with conflicts.  Region also works the same - *Canada blocks people from US, so you make a Canadian account and then it works etc. (no vpn/proxy).

**Comment by 0xboobface** [comment]
What a stupid feature :rofl: Then you should try running ctbrec in client/server mode.

**Comment by Asynchr0nous** [comment]
@z3rofeels ( using your example) how exactly do you think chaturbate knows a person is from canada or america? if it was simply changing your gender or country from the settings then site this large wouldn't bother with it at all.

**Comment by z3rofeels** [comment]
> @z3rofeels ( using your example) how exactly do you think chaturbate knows a person is from canada or america? if it was simply changing your country from the settings then site this large wouldn't bother with it at all.

You're right and to be honest, this is more of a gender issue at the moment. I haven't run into any performers region blocking recently, but in the past there was one who blocked only a single state and all I did was make an account in a random location and I was able to go into their room. For the gender you just switch from male to female and more cams populate than you would normally see.

**Comment by Asynchr0nous** [comment]
yeah the gender want makes sense and you're totally right about it but the geo-blocking is alot more common in others country than US imo.
